spark/dl/src/main/scala/com/intel/analytics/bigdl/models/utils/ModelBroadcast.scala

```diff
@@ -89,12 +90,17 @@ class ModelBroadcast[T: ClassTag](implicit ev: TensorNumeric[T]) extends Seriali
   }
 
   private def putWeightBias(broadcastWeightBias: Array[Tensor[T]],
-    localModel: Module[T]): Unit = {
-    val localWeightBias = localModel.parameters()._1
+    localModel: Module[T], inference: Boolean = true): Unit = {
+    val (localWeightBias, localGradWeightBias) = localModel.parameters()
     var i = 0
     while (i < localWeightBias.length) {
       if (localWeightBias(i) != null) {
-        localWeightBias(i).set(broadcastWeightBias(i))
+        if (!inference) {
+          localWeightBias(i).resizeAs(broadcastWeightBias(i)).copy(broadcastWeightBias(i))
+          localGradWeightBias(i).resizeAs(broadcastWeightBias(i))
```

Comment:
We can do this in DistriOptimizer, instead of ModelBroadcast, yes?

Reply:
It's not easy to init gradWeight and gradBias in DistriOptimizer....